ansible/roles/test/files/ptftests/IP_decap_test.py

```diff
@@ -345,7 +345,12 @@ def run_encap_combination_test(self, outer_pkt_type, inner_pkt_type):
         else:
             raise Exception('ERROR: Invalid inner packet type passed: ', inner_pkt_type)
 
-        for ip_range in ip_ranges:
+        if len(ip_ranges) > 150:
+            covered_ip_ranges = ip_ranges[:100] + random.sample(ip_ranges[100:], 50)  # Limit test execution time
```

Comment:
Just curious , why not ```random.sample(ip_range, 150)```?

Reply:
The IP ranges in the front of the list are kind of special, for example:
```
0.0.0.0 - 0.255.255.255
1.0.0.0 - 9.255.255.255
10.0.0.0 - 10.0.0.1
10.0.0.2 - 10.0.0.3
...
11.0.0.0 - 100.1.0.0
100.1.0.2 - 100.1.0.2
...
100.1.0.33 - 126.255.255.255
128.0.0.0 - 169.253.255.255
...
169.254.0.0 - 169.254.255.255
169.255.0.0 - 192.167.255.255 
192.168.0.0 - 192.168.0.127
...
192.168.0.128 - 192.168.0.255
...
```
So, it should be better to cover the first 100 IP ranges. Then sample 50 rest of the ranges that are of the same type. In this way, the coverage is not compromised.